pool/pool.go

```diff
@@ -1206,10 +1204,6 @@ func (p *innerPool) connection() (*clientPack, error) {
 	return nil, errors.New("no healthy client")
 }
 
-func (p *Pool) OwnerID() *user.ID {
-	return &p.sessionOwner
-}
-
```

Comment:
Could you explain for what need? The key is provided during `Pool` construction, so corresponding user ID can be calculated by caller.

Reply:
Apps can calculate user.ID without this function, indeed. 

It was just a nice way to avoid private key usage at object creation, e.g. when HTTP gateway uploads object into container, it should specify OwnerID of the object and here it is.

Probably it had more use when ephimerical key was generated inside the pool, but it is not the case right now.

Reply:
I found more than 20 use cases total in `rclone`, `restic`, `distribution`, `sftp-gw` `cdn-billing`, `cdn-tool` and `http-gw` repositories. Don't see an issue if we keep it as some syntax sugar for `pool`.

Reply:
This behavior:

1. `PrivateKey` -> `Pool`
2. `user.ID` <- `Pool`
3. `user.ID` to smth -> `Pool`

seems pretty strange: 

- why do we move one entity back and forth?
- what if we need to set `user.ID` which can't be calculated from the key?

> I found more than 20 usecases

> used for set owner id for objects and containers during creation

So we can consider `Pool` as

1. a simple intermediate layer between applications and `Client`. Here `Pool` knows nothing about `user.ID` and ID is stored on app-side.
2. a network participant who can, incl. create objects, containers, etc. In this case `Pool` can set its own `user.ID` (if it hasn't already been set) right during the operation, so we `Pool` encapsulate user ID.

1st reduces business logic in `Pool`, and allows application to control user IDs in created entities. 2nd saves current behavior, but makes it more clean in terms of OOP.

Reply:
We consider pool as 1) simple intermediate layer.

Pool contains a key, which can be transformed into `user.ID` and it was convenient to fetch key / `user.ID` from pool instead of passing it along pool instance (syntax sugar). But we can live without it.

Reply:
Current changes follow 1).